src/types/store.ts

```diff
@@ -172,7 +170,7 @@ export interface Store<
    *
    * @returns The current state tree of your application.
    */
-  getState(): S
+  getState(): ExtendState<S, StateExt>
```

Comment:
Ideally the `Store` type would only need to know about `S` and `A`. Unfortunately this is necessary since `replaceReducer` should be able to take the reducer without `StateExt` being applied (the same reducer that was passed into `createStore`). My preference would be to remove `StateExt` altogether since I don't think there's a good practical use-case for it, but short of that, this seems like the necessary solution.